app/Providers/ErrorReportingProvider.php

```diff
@@ -0,0 +1,156 @@
+    {
+        // don't report when:
+        Flare::filterExceptionsUsing(function (\Exception $e) {
+            if (! Config::get('reporting.error')) {
+                return false;
+            }
+
+            // Only run in production
+            if (! $this->app->isProduction()) {
+                return false;
+            }
+
+            // Check if git installation
+            if (! Git::repoPresent()) {
+                return false;
+            }
+
+            // Repo url must be offical one
+            if (! Str::contains(Git::remoteUrl(), ['git@github.com:librenms/librenms.git', 'https://github.com/librenms/librenms.git'])) {
+                return false;
+            }
+
+            // Check if repo is modified
+            if (! Git::unchanged()) {
+                return false;
+            }
+
+            // Check if repo is modified
+            if (! Git::officalCommit()) {
+                return false;
+            }
+
+            return true;
+        });
+
+        Flare::determineVersionUsing(function () {
+            return \LibreNMS\Util\Version::VERSION;
+        });
+
+        Flare::registerMiddleware(function (Report $report, $next) {
+
+            // Filter some extra fields for privacy
+            // Move to header middleware when switching to spatie/laravel-ignition
+            try {
+                $report->setApplicationPath('');
+                $context = $report->allContext();
+
+                if (isset($context['request']['url'])) {
+                    $context['request']['url'] = str_replace($context['headers']['host'] ?? '', 'librenms', $context['request']['url']);
+                }
+
+                if (isset($context['session']['_previous']['url'])) {
+                    $context['session']['_previous']['url'] = str_replace($context['headers']['host'] ?? '', 'librenms', $context['session']['_previous']['url']);
+                }
+
+                $context['headers']['host'] = null;
+                $context['headers']['referer'] = null;
+
+                $report->userProvidedContext($context);
+            } catch (\Exception $e) {
+            }
+
+            // Add more LibreNMS related info
+            try {
+                $version = Version::get();
+
+                $report->group('LibreNMS', [
+                    'Git version' => $version->local(),
+                    'App version' => Version::VERSION,
+                ]);
+
+                $report->group('Tools', [
+                    'Database' => $version->databaseServer(),
+                    'Net-SNMP' => $version->netSnmp(),
+                    'Python' => $version->python(),
+                    'RRDtool' => $version->rrdtool(),
+
+                ]);
+            } catch (\Exception $e) {
+            }
+
+            return $next($report);
+        });
+
+        // Override the Laravel error handler
+        $this->laravelErrorHandler = set_error_handler([$this, 'handleError']);
```

Comment:
> Returns the previously defined error handler (if any).

Oh, this is weird syntax.

Reply:
It is but, it did prevent me from having to duplicate the default Laravel handler code or do some VERY strange stuff.